apps/ledger-live-desktop/src/renderer/screens/exchange/index.tsx

```diff
@@ -33,6 +38,24 @@ const LiveAppExchange = ({ appId }: { appId: string }) => {
   const manifest = localManifest || mockManifest || remoteManifest;
   const themeType = useTheme().colors.palette.type;
 
+  // Use wallet-api ids when manifest is using apiVersion 2
+  const WALLET_API_VERSION = 2;
+  const apiVersion = manifest?.apiVersion;
+  const apiVersionMatch =
+    !!apiVersion &&
+    (apiVersion.match(/\d+|\d+\b|\d+(?=\w)/g) || []).shift() === `${WALLET_API_VERSION}`;
+  if (urlParams && apiVersionMatch) {
+    const { account: accountId } = urlParams;
+    const account = accounts.find(a => a.id === accountId);
+    if (account) {
+      const parentAccount = isTokenAccount(account)
+        ? getParentAccount(account, accounts)
+        : undefined;
+      const walletApiId = accountToWalletAPIAccount(account, parentAccount)?.id;
+      urlParams.account = walletApiId;
+    }
+  }
+
```

Comment:
I would have put this in a `useEffect`

Also it would be better to use our constant for `WALLET_API_VERSION` and check if the manifest satisfies the version using the semver package like we do here: https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-mobile/src/components/Web3AppWebview/index.tsx#L10-L11
```tsx
import semver from "semver";
import { WALLET_API_VERSION } from "@ledgerhq/live-common/wallet-api/constants";

if (semver.satisfies(WALLET_API_VERSION, manifest.apiVersion)) {
  // your logic
}
```